services/src/main/java/org/sdo/demo/AioApiServlet.java

```diff
@@ -427,10 +442,8 @@ public void putAsync(AsyncContext asyncCtx) {
         final String fileName = list.get(4);
         File guidDir = Paths.get(fsRootDir, fsDevicesDir, uuid).toFile();
         Path filePath = Paths.get(fsRootDir, fsDevicesDir, uuid, fileName);
-        if (!guidDir.exists()) {
-          guidDir.mkdir();
-        }
-        if (checkFileNameSanity(fileName) && pathTraversalCheck(filePath)) {
+        if (checkFileNameSanity(fileName) && pathTraversalCheck(filePath)
+            && guidDir.exists() && checkGuid(uuid)) {
```

Comment:
In this PR, don't restrict creation of new GUID directory. So create the guid directory if it doesn't exist, but move the check inside the if loop.

if (checkFileNameSanity(fileName) && pathTraversalCheck(filePath)
            && checkGuid(uuid)) {
        if (!guidDir.exists()) {
          guidDir.mkdir();
        }
       copyFile(req, res, filePath);
    } else {